sagemaker_processing/local_pyspark/local_pyspark_example.ipynb

```diff
@@ -0,0 +1,499 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run PySpark locally on SageMaker Studio\n",
+    "\n",
+    "This notebook shows you how to run PySpark code locally within a SageMaker Studio notebook. The dependencies are installed in the notebok, so you can run this notebook on any image/kernel, including BYO images. For this example, you can choose the Data Science image and Python 3 kernel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# setup - install JDK\n",
+    "# you only need to run this once per kernel\n",
+    "%conda install openjdk -y"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# install PySpark\n",
+    "%pip install pyspark==3.2.0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# upgrade sagemaker sdk\n",
+    "%pip install --upgrade sagemaker"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import PySpark and build Spark session\n",
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "spark = (\n",
+    "    SparkSession.builder.appName(\"PySparkApp\")\n",
+    "    .config(\"spark.jars.packages\", \"org.apache.hadoop:hadoop-aws:3.2.2\")\n",
+    "    .config(\n",
+    "        \"fs.s3a.aws.credentials.provider\",\n",
+    "        \"com.amazonaws.auth.ContainerCredentialsProvider\",\n",
```

Comment:
Let's add a cell describing when to use ContainerCred (Studio) vs. InstanceCred (Classic Notebooks)

Reply:
I've added it as a callout under the cell-
> 2. If you are running this notebook in a SageMaker Studio notebook, run the above cell as-is. If you are running on a SageMaker notebook instance, replace `com.amazonaws.auth.ContainerCredentialsProvider` with `com.amazonaws.auth.InstanceProfileCredentialsProvider`.